core/src/com/unciv/ui/pickerscreens/ImprovementPickerScreen.kt

```diff
@@ -23,6 +24,17 @@ class ImprovementPickerScreen(
     private val unit: MapUnit,
     private val onAccept: ()->Unit,
 ) : PickerScreen() {
+
+    companion object {
+        val specialCasedProblems = setOf(
+            TileInfo.ImprovementBuildingProblem.MissingTech,
+            TileInfo.ImprovementBuildingProblem.NotJustOutsideBorders,
+            TileInfo.ImprovementBuildingProblem.OutsideBorders,
+            TileInfo.ImprovementBuildingProblem.MissingResources)
+
+        fun isHopelesslyUnbuildable(problems: List<TileInfo.ImprovementBuildingProblem>?): Boolean = problems != null && problems.any { it !in specialCasedProblems }
+    }
```

Comment:
`enum class ImprovementBuildingProblem(val isHopeless: Boolean = false)`?

Reply:
I disagree. It has little to do with game mechanics, it doesn't affect anything. Instead, it has relevance only to UI: if the picker thinks the improvement is important to show to the user, and knows how to do that. For example `UnmetConditional` might also be a non-fatal problem (what is it, btw?), but since the picker doesn't know what to do about it, it is treated as such.

Renamed `specialCasedProblem` to `reportableProblems` and documented it to make my reasoning for placing it in `ImprovementPickerScreen` clearer.

Reply:
Also, to make my position clearer. I think that e.g. `TileInfo.toMarkup()` is misplaced. I would rather move it to UI code, as it's not really about logic, only about how things get shown to the player.

Reply:
> e.g. TileInfo.toMarkup() is misplaced

True, from a certain viewpoint. I placed it there as it was a variation of getCivilopediaText (same argument applies) which was a variation of getDescription() that I found there. And sometimes, a little, I get that warm fuzzy feeling if code that accesses a lot of an Object's properties but few to none of other Object's properties lives where it has the most direct access to those fields, and sometimes allows leaving members private... Feel free to clean up, but please separately? Also, expect discussions for such a cleanup. And - reworking all of the Civilopedia "rendering" related stuff will be better mergeable in small doses. 

Lastly, I have somewhere there in the back of my mind the idea that some text that is currently rendered as simple text would look far better and more informative (not to mention the usefulness of linkability). Like: the tech researched alert's info on what the tech does, or in some pickers where you select a RulesetObject and it displays the RulesetObject.getDescription() down there in descriptionLabel. When someone gets around to do that, there may be small changes necessary to these functions, a getDescription might be left without callers... You could if you wish try this idea on some parts and combine with your cleanup vision... Or not.

> reportableProblems and documented

Sounds perfect - sorry for not yet taking the time to check your work directly (so far I just read the mail). I distract myself too easily.

Reply:
> Feel free to clean up, but please separately?

I'm certainly not going to do that in this PR, I was just illustrating why I consider placement of `reportableProblems` in UI code correct by pointing out what I think is wrong as a counterexample. Not that I want to do something about that immediately (or at all).